microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/model/parser/PropertiesHandler.java

```diff
@@ -122,4 +122,22 @@
 
 	void delimiterAssign(ParseContext context);
 
+	/**
+	 * Returns true if property name exists and end offset
+	 * is missing. Returns false otherwise
+	 * 
+	 * @return true if property name exists and end offset
+	 * is missing. Returns false otherwise
+	 */
+	boolean propertyNameEndMissing();
+
+	/**
+	 * Returns true if property value exists and end offset
+	 * is missing. Returns false otherwise.
+	 * 
+	 * @return true if property value exists and end offset
+	 * is missing. Returns false otherwise.
+	 */
+	boolean propertyValueEndMissing();
+

```

Comment:
I decided to add these methods to `PropertiesHandler` in order to determine if the parser was done parsing the name/value.

If `propertyNameEndMissing()` or `propertyValueEndMissing()` returns true, this would mean that the name/value was not finished being parsed, and must continue parsing the name/value on the next line.

Reply:
Why have you done this choice? The idea of the handler is the same idea than SAX handler, just collect some events. 

It's strange to delegate the propertyNameEndMissing behavior to the handler implementation. 

IMHO, It should be cleaner to manage this case inside the PropertiesParser. But perhaps it's hard to do that?

Reply:
@xorye I think we should have 2 methods :

 * getText(boolean skipMultiLine)
 * getText() which call getText(false)

because getText() is used too for formatting. I suspect that you should have a trouble with https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/services/MicroProfileFormatter.java#L155 when you will try to format property key in multi line (you will loose multi line, I think)

Separate into to methods will give the capability to use getText in different context 

 * for getPropertyName, \n must be escaped
 * for formatting it should not.

Reply:
My latest push fixes this by managing the cases inside PropertiesParser. I now use this enum to keep track of what needs to be parsed on the next line:
https://github.com/redhat-developer/quarkus-ls/blob/3e4c5b42ed8db0c97aa16211ef6ecca61ad9c691/microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/model/parser/PropertiesParser.java#L45-L52